File: src/Libraries/c/mylib.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

__declspec(dllexport) char *concatStrings(const char *str1, const char *str2)
{
    size_t lenStr1 = strlen(str1);
    size_t lenStr2 = strlen(str2);
    char *result = (char *)malloc(lenStr1 + lenStr2 + 1); // +1 para el terminador nulo
    if (result == NULL)
    {
        return NULL; // Fallo en la asignación de memoria
    }
    snprintf(result, (lenStr1 + lenStr2 + 1), "%s%s", str1, str2);
    return result;
}

__declspec(dllexport) void free_memory(char *ptr)
{
    free(ptr);
}

__declspec(dllexport) int add(int a, int b)
{
    return (a + b);
}

File: src/Libraries/zig/mylib.zig
const std = @import("std");

export fn add(a: i32, b: i32) i32 {
    return a + b;
}

pub export fn concatStrings(c_str1: [*c]u8, c_str2: [*c]u8) [*c]u8 {
    const allocator = std.heap.c_allocator;

    const result = std.fmt.allocPrint(allocator, "{s}{s}", .{ c_str1, c_str2 }) catch return null;

    return result.ptr;
}

fn c_strlen(s: [*c]u8) usize {
    var len: usize = 0;
    while (s[len] != 0) {
        len += 1;
    }
    return len;
}

pub export fn free_memory(ptr: [*c]u8) void {
    const allocator = std.heap.c_allocator;
    const len = c_strlen(ptr);
    allocator.free(ptr[0..len]);
}
